src/DotLiquid/Variable.cs

```diff
@@ -1,141 +1,144 @@
-                            string filterName = filterNameMatch.Groups[1].Value;
-                            List<string> filterArgs = R.Scan(f, FilterArgRegex);
-                            Filters.Add(new Filter(filterName, filterArgs.ToArray()));
-                        }
-                    }
-                }
-            }
-        }
-
-        public void Render(Context context, TextWriter result)
-        {
-            object output = RenderInternal(context);
-
-            if (output is ILiquidizable)
-                output = null;
-
-            if (output != null)
-            {
-                var transformer = Template.GetValueTypeTransformer(output.GetType());
-
-                if(transformer != null)
-                    output = transformer(output);
-
-                string outputString;
-                if (output is IEnumerable)
-#if NET35
-                    outputString = string.Join(string.Empty, ((IEnumerable)output).Cast<object>().Select(o => o.ToString()).ToArray());
-#else
-                    outputString = string.Join(string.Empty, ((IEnumerable)output).Cast<object>());
-#endif
-                else if (output is bool)
-                    outputString = output.ToString().ToLower();
-                else
-                    outputString = output.ToString();
-                result.Write(outputString);
-            }
-        }
-
-        private object RenderInternal(Context context)
-        {
-            if (Name == null)
-                return null;
-
-            object output = context[Name];
-
-            Filters.ToList().ForEach(filter =>
-            {
-                List<object> filterArgs = filter.Arguments.Select(a => context[a]).ToList();
-                try
-                {
-                    filterArgs.Insert(0, output);
-                    output = context.Invoke(filter.Name, filterArgs);
-                }
-                catch (FilterNotFoundException ex)
-                {
-                    throw new FilterNotFoundException(string.Format(Liquid.ResourceManager.GetString("VariableFilterNotFoundException"), filter.Name, _markup.Trim()), ex);
-                }
-            });
-
-            if (output is IValueTypeConvertible)
-                output = ((IValueTypeConvertible) output).ConvertToValueType();
-
-            return output;
-        }
-
-        /// <summary>
-        /// Primarily intended for testing.
-        /// </summary>
-        /// <param name="context"></param>
-        /// <returns></returns>
-        internal object Render(Context context)
-        {
-            return RenderInternal(context);
-        }
-
-        public class Filter
-        {
-            public Filter(string name, string[] arguments)
-            {
-                Name = name;
-                Arguments = arguments;
-            }
-
-            public string Name { get; set; }
-            public string[] Arguments { get; set; }
-        }
-    }
-}
+using System.Collections;
```

Comment:
how i "love" Bill Gates for makeing line separator two characters, I'll adjust that